test/suite/test_oligarch12.py

```diff
@@ -0,0 +1,115 @@
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, time, wiredtiger, wttest
+from helper_disagg import DisaggConfigMixin
+
+# test_oligarch12.py
+#    Pick up different checkpoints.
+class test_oligarch12(wttest.WiredTigerTestCase, DisaggConfigMixin):
+    nitems = 500
+
+    conn_base_config = 'oligarch_log=(enabled),statistics=(all),statistics_log=(wait=1,json=true,on_close=true),' \
+                     + 'disaggregated=(stable_prefix=.,page_log=palm),'
+    conn_config = conn_base_config + 'disaggregated=(role="leader")'
+
+    create_session_config = 'key_format=S,value_format=S'
+
+    uri = "oligarch:test_oligarch12"
+
+    # Load the page log extension, which has object storage support
+    def conn_extensions(self, extlist):
+        if os.name == 'nt':
+            extlist.skip_if_missing = True
+        extlist.extension('page_log', 'palm')
+        self.pr(f"{extlist=}")
+
+    # Custom test case setup
+    def early_setup(self):
+        os.mkdir('follower')
+        # Create the home directory for the PALM k/v store, and share it with the follower.
+        os.mkdir('kv_home')
+        os.symlink('../kv_home', 'follower/kv_home', target_is_directory=True)
+
+    # Test inserting records into a follower that turned into a leader
+    def test_oligarch12(self):
+        self.session.create(self.uri, self.create_session_config)
+
+        conn_follow = self.wiredtiger_open('follower', self.extensionsConfig() + ',create,' + self.conn_base_config + 'disaggregated=(role="follower")')
+        session_follow = conn_follow.open_session('')
+        session_follow.create(self.uri, self.create_session_config)
+
+        value1 = "aaa"
+        value2 = "bbb"
+
+        # Create version 1 of the data
+        cursor = self.session.open_cursor(self.uri, None, None)
+        for i in range(self.nitems):
+            cursor[str(i)] = value1
+            if i % 250 == 0:
+                time.sleep(1)
+        cursor.close()
+        time.sleep(1)
+        self.session.checkpoint()
+        time.sleep(1)
+        checkpoint1 = self.disagg_get_complete_checkpoint()
+
+        # Create version 2 of the data
+        cursor = self.session.open_cursor(self.uri, None, None)
+        for i in range(self.nitems):
+            if i % 10 == 0:
+                cursor[str(i)] = value2
+            if i % 250 == 0:
+                time.sleep(1)
+        cursor.close()
+        time.sleep(1)
+        self.session.checkpoint()
+        time.sleep(1)
+        checkpoint2 = self.disagg_get_complete_checkpoint()
+
+        # Pick up the first version and check
+        conn_follow.reconfigure(f'disaggregated=(checkpoint_id={checkpoint1})')
+        cursor = session_follow.open_cursor(self.uri, None, None)
+        for i in range(self.nitems):
+            self.assertEquals(cursor[str(i)], value1)
+        cursor.close()
+
+        self.skipTest('The test will fail due to unaligned buffers during connection close')
```

Comment:
If the test is skipped, how do you know the new code changes work?  Does the test get further than it did without the changes?

Reply:
Without the palm changes in the PR, the code would fail in the code block above this line.

If I comment out the skip, the next code block would work, but then the test will fail in connection close - this is the same issue as in test_oligarch07.